File: api/HomeBudget.API/HomeBudget.Application/DTOs/TransacaoDtos/UpdateTransacaoDto.cs
using HomeBudget.Domain.Enums;

namespace HomeBudget.Application.DTOs.TransacaoDtos
{
    public class UpdateTransacaoDto
    {
        public Guid Id { get; set;  }
        public required Guid IdPessoa { get; set; }
        public required Guid IdCategoria { get; set; }
        public Tipo Tipo { get; set; }

        public required string Nome { get; set;  }
        public required string Descricao { get; set; } = string.Empty;


        public double Receitas { get; set; } = 0.00;
        public double Despesas { get; set; } = 0.00;
        public double Saldo { get; set; } = 0.00;
    }
}

File: web/src/interfaces/TransacaoDtos/UpdateTransacaoDto.ts
import type { TipoTransacao } from "../../enums/TipoTransacaoEnum";

export class UpdateTransacaoDto {
  id: string;
  idPessoa: string;
  idCategoria: string;
  tipo: TipoTransacao;
  nome: string;
  descricao: string;
  receitas: number;
  despesas: number;
  saldo: number;

  constructor(
    id: string,
    idPessoa: string,
    idCategoria: string,
    tipo: TipoTransacao,
    nome: string,
    descricao: string,
    receitas: number = 0.0,
    despesas: number = 0.0,
    saldo: number = 0.0
  ) {
    this.id = id;
    this.idPessoa = idPessoa;
    this.idCategoria = idCategoria;
    this.tipo = tipo;
    this.nome = nome;
    this.descricao = descricao;
    this.receitas = receitas;
    this.despesas = despesas;
    this.saldo = saldo;
  }
}